src/mlpack/core/math/random.hpp

```diff
@@ -78,6 +78,44 @@ inline int RandInt(const int lo, const int hiExclusive)
 }
 
 /**
+ * Generates a random integer from range [0, N) with given probabilities
+ * P(0), P(1), ..., P(N-1) (arbitrary discrete distribution).
+ * 
+ * @param weights Probabilities of individual numbers. Note that the function
+ *                accepts unnormalized probabilties as long as they are
+ *                non-negative and sum to a positive number.
+ * @return A random integer sampled from the specified distribution.
+ */
+inline int RandInt(const arma::vec& weights)
+{
+  // Build cumulative probabilities from event probabilities.
+  std::vector<double> slots(weights.size());
+  double runningSum = 0;
+  for (size_t i = 0; i < weights.size(); ++i) {
```

Comment:
Well, we still need to run that check - just to alert user if he tries to use it in the wrong way (intentialloy or otherwise)